reference/configuration/security.rst

```diff
@@ -22,19 +22,376 @@ key in your application configuration.
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Using the `PBKDF2`_ encoder is no longer recommended since PHP added support for
+Argon2i and bcrypt. Legacy application still using it are encouraged to upgrade
+to those newer encoding algorithms.
+
+firewalls
+---------
+
+This is arguably the most important option of the security config file. It
+defines the authentication mechanism used for each URL (or URL pattern) of your
+application:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # app/config/security.yml
+
+        security:
+            # ...
+            firewalls:
+                # 'main' is the name of the firewall (can be chosen freely)
+                main:
+                    # 'pattern' is a regular expression matched against the incoming
+                    # request URL. If there's a match, authentication is triggered
+                    pattern: ^/admin
+                    # the rest of options depend on the authentication mechanism
+                    # ...
+
+    .. code-block:: xml
+
+        <!-- app/config/security.xml -->
+        <?xml version="1.0" encoding="UTF-8"?>
+        <srv:container xmlns="http://symfony.com/schema/dic/security"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xmlns:srv="http://symfony.com/schema/dic/services"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+            <config>
+                <!-- ... -->
+
+                <!-- 'pattern' is a regular expression matched against the incoming
+                     request URL. If there's a match, authentication is triggered -->
+                <firewall name="main" pattern="^/admin">
+                    <!-- the rest of options depend on the authentication mechanism -->
+                    <!-- ... -->
+                </firewall>
+            </config>
+        </srv:container>
+
+    .. code-block:: php
+
+        // app/config/security.php
+
+        // ...
+        $container->loadFromExtension('security', [
+            'firewalls' => [
+                // 'main' is the name of the firewall (can be chosen freely)
+                'main' => [
+                    // 'pattern' is a regular expression matched against the incoming
+                    // request URL. If there's a match, authentication is triggered
+                    'pattern' => '^/admin',
+                    // the rest of options depend on the authentication mechanism
+                    // ...
+                ],
+            ],
+        ]);
+
+.. tip::
+
+    Read :doc:`this article </security/firewall_restriction>` to learn about how
+    to restrict firewalls by host and HTTP methods.
+
+In addition to some common config options, the most important firewall options
+depend on the authentication mechanism, which can be any of these:
+
+.. code-block:: yaml
+
+    # app/config/security.yml
+
+        security:
```

Comment:
There should not be an empty line between the file name comment and the code. And also, it looks like the syntax is indented 4 spaces to much?

Reply:
Actually, the indent seems to be from a configuration block. Shall we add the XML and PHP examples as well?

Reply:
Fixed the indent and removed the blank line.